jaeger-zipkin/README.md

```diff
@@ -0,0 +1,32 @@
+# jaeger-zipkin
+These components make Jaeger compatible with Zipkin instrumentation and transports.
+
+For example, you can join traces with other Zipkin (B3) instrumented applications (such as Brave), and send spans to
+Zipkin on standard transports such as Http or Kafka.
+
+# Usage
+To make Jaeger Zipkin-compatible, you need to register injectors and extractors, and configure a remote reporter.
+
+## B3 Propagation
+Zipkin systems most often use B3 propagation. This includes setting and reading headers like "X-B3-TraceId". Once you
+register `B3TextMapCodec`, Jaeger can join traces started by other Zipkin instrumented applications.
+
+For example:
+```java
+b3Codec = new B3TextMapCodec();
+tracer = new Tracer.Builder(serviceName, reporter, sampler)
+                   .registerInjector(Format.Builtin.HTTP_HEADERS, b3Codec)
+                   .registerExtractor(Format.Builtin.HTTP_HEADERS, b3Codec)
+                   ...
+```
+
+## Sending data to Zipkin
+Zipkin supports transports including Http and Kafka. You can configure Jaeger to send to a Zipkin server with
+`ZipkinSender`.
+
+For example:
+```java
+reporter = new RemoteReporter(ZipkinSender.create("http://localhost:9411/api/v1/spans"));
+tracer = new Tracer.Builder(serviceName, reporter, sampler)
+                   ...
```

Comment:
This is actually linked to
https://github.com/openzipkin/zipkin-java-reporter which brave will use in
the future. It has its own kafka reporter (and will soon have a scribe
one). The main goal of zipkin-java-reporter was to allow people to share
code (like for OT reporters), but without needing to bring in dependencies
on a different instrumentation project.

On Fri, Aug 26, 2016 at 11:00 PM, Yuri Shkuro <notifications@github.com>
wrote:

> In jaeger-zipkin/README.md
> <https://github.com/uber/jaeger-client-java/pull/34#discussion_r76433413>:
>
> > +b3Codec = new B3TextMapCodec();
> > +tracer = new Tracer.Builder(serviceName, reporter, sampler)
> > +                   .registerInjector(Format.Builtin.HTTP_HEADERS, b3Codec)
> > +                   .registerExtractor(Format.Builtin.HTTP_HEADERS, b3Codec)
> > +                   ...
> > +```
> > +
> > +## Sending data to Zipkin
> > +Zipkin supports transports including Http and Kafka. You can configure Jaeger to send to a Zipkin server with
> > +`ZipkinSender`.
> > +
> > +For example:
> > +```java
> > +reporter = new RemoteReporter(ZipkinSender.create("http://localhost:9411/api/v1/spans"));
> > +tracer = new Tracer.Builder(serviceName, reporter, sampler)
> > +                   ...
>
> Q: because this is linked to Brave, does it mean we can reuse its Kafka
> transport?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/uber/jaeger-client-java/pull/34/files/ad4cc1dcf39b7681274c424b28f26a5549cad88a#r76433413>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD619wvrEPfowvt9m1pJxKLwdaA8dBXks5qjv-TgaJpZM4Jt4ou>
> .
>